src/index.ts

```diff
@@ -170,441 +155,196 @@ export class ERC725 {
+                .then((buffer) => new Uint8Array(buffer));
             }
 
-            return a.json();
+            return response.json();
           });
         } catch (error) {
-          console.error(error, `GET request to ${result.url} failed`);
+          console.error(error, `GET request to ${dataEntry.url} failed`);
           throw error;
         }
 
-        return response &&
-          this.hashAndCompare(response, result.hash, lowerCaseHashFunction)
-          ? response
-          : null;
-      }
-      default:
-        return result;
-    }
-  }
+        const isDataAuthentic = hashAndCompare(
+          receivedData,
+          dataEntry.hash,
+          dataEntry.hashFunction,
+        );
 
-  /**
-   * @param data An object of keys matching to corresponding schema element names, with associated data.
-   * @returns all encoded data as per required by the schema and provided data
-   *
-   * **Example**
-   *
-   * ```javascript
-   * myERC725.encodeAllData({
-   *   LSP3Profile: {
-   *     hashFunction: 'keccak256(utf8)',
-   *     hash: '0x820464ddfac1bec070cc14a8daf04129871d458f2ca94368aae8391311af6361',
-   *     url: 'ifps://QmYr1VJLwerg6pEoscdhVGugo39pa6rycEZLjtRPDfW84UAx'
-   *   },
-   *   'LSP3IssuedAssets[]': [
-   *     '0xD94353D9B005B3c0A9Da169b768a31C57844e490',
-   *     '0xDaea594E385Fc724449E3118B2Db7E86dFBa1826'
-   *   ],
-   *   LSP1UniversalReceiverDelegate: '0x1183790f29BE3cDfD0A102862fEA1a4a30b3AdAb'
-   * });
-   *
-   * // > [
-   * //  {
-   * //      "key": "0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5",
-   * //      "value": "0x6f357c6a820464ddfac1bec070cc14a8daf04129871d458f2ca94368aae8391311af6361696670733a2f2f516d597231564a4c776572673670456f73636468564775676f3339706136727963455a4c6a7452504466573834554178"
-   * //  },
-   * //  {
-   * //      "key": "0x0cfc51aec37c55a4d0b1a65c6255c4bf2fbdf6277f3cc0730c45b828b6db8b47",
-   * //      "value": "0x1183790f29be3cdfd0a102862fea1a4a30b3adab"
-   * //  },
-   * //  {
-   * //      "key": "0x3a47ab5bd3a594c3a8995f8fa58d0876c96819ca4516bd76100c92462f2f9dc0",
-   * //      "value": "0x0000000000000000000000000000000000000000000000000000000000000002"
-   * //  },
-   * //  {
-   * //      "key": "0x3a47ab5bd3a594c3a8995f8fa58d087600000000000000000000000000000000",
-   * //      "value": "0xd94353d9b005b3c0a9da169b768a31c57844e490"
-   * //  },
-   * //  {
-   * //      "key": "0x3a47ab5bd3a594c3a8995f8fa58d087600000000000000000000000000000001",
-   * //      "value": "0xdaea594e385fc724449e3118b2db7e86dfba1826"
-   * //  }
-   * // ]
-   * ```
-   * :::note Try it
-   * https://stackblitz.com/edit/erc725js-encode-all-data?devtoolsheight=66&file=index.js
-   * :::
-   */
-  encodeAllData(data) {
-    return encodeAllData(this.options.schema, data);
+        if (!isDataAuthentic) {
+          accumulator[key] = null;
+        }
+
+        accumulator[key] = receivedData;
+        return accumulator;
+      }, {});
```

Comment:
Ah just saw the function `hashAndCompare` should throw if hash mismatchs?

Then in that case, `isDataAuthentic` will always be `true` at this stage? (Otherwise it will throw before)

Maybe you want to add a try / catch at line with `hashAndCompare` ?

Then the condition can be "used" correctly?

Reply:
How about this? Should I save the result of the function call in a variable for readability? 

```ts
        accumulator[key] = isDataAuthentic(
          receivedData,
          dataEntry.hash,
          dataEntry.hashFunction,
        )
          ? receivedData
          : null;
```